libdrgn/dwarf_index.c

```diff
@@ -2677,6 +2798,17 @@ drgn_dwarf_index_iterator_next(struct drgn_dwarf_index_iterator *it,
 		}
 	}
 
+	if (resolve_ns && die->tag == DW_TAG_namespace)
```

Comment:
I think we should be indexing the namespace when it is first passed to `drgn_dwarf_index_iterator_init()` (i.e., the first time we actually want to access it) rather than here. `drgn_dwarf_index_iterator_init()` would need to return `struct drgn_error *`, but that's an easy change.